src/tlo/analysis/life_expectancy.py

```diff
@@ -0,0 +1,305 @@
+
+    # get children aged=0 in target period
+    num_children = extract_results(
+        results_folder,
+        module="tlo.methods.demography",
+        key="num_children",
+        custom_generate_series=extract_infants_during_target_period,
+        do_scaling=False
+    )
+    num_f0 = num_children / 2  # todo should this be exactly 50/50?
+    num_m0 = num_children / 2
+
+    # get age_range_m and age_range_f for all age-groups
+    num_males = _extract_pop_during_target_period(results_folder=results_folder, key='age_range_m')
+    num_females = _extract_pop_during_target_period(results_folder=results_folder, key='age_range_f')
+    num_males = num_males.reset_index(level=0, drop=True)
+    num_females = num_females.reset_index(level=0, drop=True)
+
+    # then subtract age 0 from age-group 0-4
+    num_males_adj = num_males.iloc[0:1, :] - num_m0.iloc[0:1, :].values
+    # add adjusted data back into dataframe
+    num_males.iloc[0] = num_males_adj.values
+    # change first row index
+    num_males.index.values[0] = '1-4'
+    # add in infant numbers to main dataframe
+    t1 = pd.concat([num_m0, num_males])
+    # set correct index
+    t1.index.values[0] = '0'
+
+    # repeat for females
+    num_females_adj = num_females.iloc[0:1, :] - num_f0.iloc[0:1, :].values
+    num_females.iloc[0] = num_females_adj.values
+    num_females.index.values[0] = '1-4'
+    t2 = pd.concat([num_f0, num_females])
+    t2.index.values[0] = '0'
+
+    # combine pop aged 90+
+    # Sum the last three rows column-wise
+    last_three_rows_sum = t1.iloc[-3:].sum()
+    t1.loc['90+'] = last_three_rows_sum
+
+    last_three_rows_sum = t2.iloc[-3:].sum()
+    t2.loc['90+'] = last_three_rows_sum
+
+    # remove unneeded columns
+    indices_to_remove = ['90-94', '95-99', '100+']
+    t1 = t1.drop(indices_to_remove, axis=0)
+    t2 = t2.drop(indices_to_remove, axis=0)
+
+    # output is 0, 1-4, 5-9, 10-14 and index by male/female
+    # join male and female dataframe and add index for sex
+    t3 = pd.concat([t1, t2], ignore_index=True)
+
+    # Create a MultiIndex for rows using age group and 'Male' and 'Female'
+    multi_index_rows_male = pd.MultiIndex.from_product([t1.index, ['M']], names=['age_group', 'sex'])
+    multi_index_rows_female = pd.MultiIndex.from_product([t1.index, ['F']], names=['age_group', 'sex'])
+    t3.index = multi_index_rows_male.append(multi_index_rows_female)
+
+    return t3
+
+
+# %% GENERATE LIFE EXPECTANCY ESTIMATES
+
+def _estimate_life_expectancy(person_years_at_risk, number_of_deaths_in_interval):
+    """
+    for a single run, estimate life expectancy for males and females
+    return: pd.Series
+    """
+
+    estimated_life_expectancy = pd.Series(dtype=float)
+
+    # first age-group is 0, then 1-4, 5-9, 10-14 etc. 22 categories in total
+    level_0_values = person_years_at_risk.index.get_level_values('age_group').unique()
+    interval_width = pd.Series([1] + [4] + [5] * 18, index=level_0_values)
```

Comment:
I notice that level_0_values is not sorted before/after using `unqiue` so we don't know the order of the values. The logic on line 176 relies on the age-groups being as defined somewhere else. So, I think its much safer to work do a sort of the values and work out the width directly what you get: e.g. using `diff`.

Reply:
I've made this change now